### PR TITLE
Return objects referenced in the processed templates

### PR DESCRIPTION
**Pull request by ckandag**
Signed-off-by: Chaitanya Kandagatla <ckandaga@redhat.com>

For Isssue https://github.com/stolostron/backlog/issues/25848

### CODE REVIEW

**Review by dhaiducek — COMMENTED** [review]
This looks good, and returning the resolved object will be helpful--is there context and/or a backing issue for this PR?

**Comment by ckandag** [comment]
> This looks good, and returning the resolved object will be helpful--is there context and/or a backing issue for this PR?

@dhaiducek , here is the related issue https://github.com/stolostron/backlog/issues/25848,  also added it to the top section in the PR

File: pkg/templates/templates_test.go [hunk 1]
```diff
@@ -529,6 +539,95 @@ func TestResolveTemplate(t *testing.T) {
 	}
 }
 
+func TestReferencedObjs(t *testing.T) {
```
Comment (dhaiducek) [inline pkg/templates/templates_test.go:542]: This is missing a test for `lookup`.

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Looks really good to me! One comment on the tests.

File: pkg/templates/templates_test.go (hunk 1 above)
Comment (ckandag) [reply pkg/templates/templates_test.go:542]: right, we have no  tests for lookup right now in other TemplateResolve tests as well, the test setup curently only for SimpleClient not dynamicClient ( dynamic resources)

Comment (dhaiducek) [reply pkg/templates/templates_test.go:542]: I see--thanks for clarifying!

**Review by dhaiducek — APPROVED** [review]

**Comment by mprahl** [comment]
Late to the party but this looks good. Thanks @ckandag!

**Comment by ckandag** [comment]
@mprahl  and @dhaiducek  : Unfortunately I did end up making an error which I realized while test changes to config-policy-controller.

The struct fields  names need to be capitalized to access outside the package.  The unit test did not catch that as they are in the same pkg.

I will make another PR for that

**Pull request merged**